roles/common/tasks/networking.yml

```diff
@@ -13,6 +13,32 @@
   with_items: network_interfaces_d
   tags: ['prereboot']
 
+- name: set slrouter address
+  set_fact: sl_router_address="{{ ansible_default_ipv4.network.split('.')[0] }}.{{ ansible_default_ipv4.network.split('.')[1] }}.{{ansible_default_ipv4.network.split('.')[2] }}.{{ (ansible_default_ipv4.network.split('.')[3] | int) + 1 }}"
+  when: ansible_default_ipv4.interface == hostvars[inventory_hostname][primary_interface].device   
+
+- name: add static route for multi subnet
+  command: "/sbin/ip route replace {{ item }} via {{ sl_router_address }}"
+  when: (not (ansible_default_ipv4.network ==  ( item | ipaddr('network') ))) and (ansible_default_ipv4.interface == hostvars[inventory_hostname][primary_interface].device)
+  with_items: "{{ undercloud_cidr }}"
+
+- name: Set network config file
+  set_fact: nw_config_file="/etc/network/interfaces"
+  when: network_interfaces is defined
+
+- name: Set network config file
```

Comment:
@paulczar 
Below will be the changes. Please comment.
- name: Set network config file
  set_fact: nw_config_file="/etc/network/interfaces"
  when: network_interfaces is defined

- name: Stat /etc/network/interfaces.d/{{ ansible_default_ipv4.interface }}.cfg
  stat: path=/etc/network/interfaces.d/{{ ansible_default_ipv4.interface }}.cfg
  register: interface_exists
  
- name: Set network config file
  set_fact: nw_config_file="/etc/network/interfaces.d/{{ ansible_default_ipv4.interface }}.cfg"
  when: network_interfaces_d is defined and interface_exists.stat.exists is defined and interface_exists.stat.exists


Check if the file /etc/network/interfaces/{{ ansible_default_ipv4.interface }}.cfg exists. If it exists, change the nw_config_file accordingly and check for the route.